File: Linkpoint/src/main/java/com/linkpoint/ui/common/FragmentActivityFactory.java
package com.lumiyaviewer.lumiya.ui.common;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;

public interface FragmentActivityFactory {
    Intent createIntent(Context context, Bundle bundle);

    Class<? extends Fragment> getFragmentClass();
}

File: Linkpoint/src/main/java/com/linkpoint/ui/common/FragmentActivityFactory.kt
package com.linkpoint.ui.common

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment

interface FragmentActivityFactory {
    fun createIntent(context: Context, bundle: Bundle): Intent
    fun getFragmentClass(): Class<out Fragment>
}